### PR TITLE
Homework/10

### PR DESCRIPTION
**Pull request by cjahermanns**
Is this what you meant with the assignment?

### CODE REVIEW

File: Homework2.Tests/ArrayExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+{
+    public class ArrayExtensionsTests
+    {
+        [Theory]
+        [InlineData(999, 4)]
+        [InlineData(999, 0)]
+        [InlineData(999, 10)]
+        public void AddAt_adds_at_idx(int item, int idx)
+        {
+            var testarray = new int[] { 11, 22, 33, 44, 55, 66, 77, 88, 99, 111 };
```
Comment (Almantask) [inline Homework2.Tests/ArrayExtensionsTests.cs]: Overkill test data. Having 3 elements would be enough.

File: Homework2.Tests/ArrayExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace Homework2.Tests
+{
+    public class ArrayExtensionsTests
+    {
+        [Theory]
+        [InlineData(999, 4)]
+        [InlineData(999, 0)]
+        [InlineData(999, 10)]
+        public void AddAt_adds_at_idx(int item, int idx)
```
Comment (Almantask) [inline Homework2.Tests/ArrayExtensionsTests.cs:19]: What happens if idx is higher than array size?

Comment (Almantask) [reply Homework2.Tests/ArrayExtensionsTests.cs:19]: What about negative?

Comment (Almantask) [reply Homework2.Tests/ArrayExtensionsTests.cs:19]: same for all methods here.

**Review by Almantask — CHANGES_REQUESTED** [review]

File: Homework2.Tests/ArrayExtensionsTests.cs (hunk 1 above)
Comment (cjahermanns) [reply Homework2.Tests/ArrayExtensionsTests.cs]: Made it smaller and used a ctor to init it.

File: Homework2.Tests/ArrayExtensionsTests.cs (hunk 2 above)
Comment (cjahermanns) [reply Homework2.Tests/ArrayExtensionsTests.cs:19]: Changed existing and added new tests. Also added exceptions to the methods (and changed the `RemoveAt(int idx)` one).

**Review by Almantask — APPROVED** [review]

**Pull request closed without merging**